Device Attestation vs. Device Assignment (#22)
thomas-fossati/draft-eat-da: draft-poirier-rats-eat-da.md

--- abstract

>  confidential computing, device assignment (DA) is the method by which a device (e.g., network adapter, GPU), whether on-chip or behind a PCIe Root Port, is assigned to a Trusted Virtual Machine (TVM).

Reviewer: ```suggestion
In confidential computing, device assignment (DA) is the method by which a device (e.g., network adapter, GPU), whether on-chip or behind a PCIe Root Port, is assigned to a Trusted Virtual Machine (TVM).
```

For the TVM to trust an assigned device, the device must provide the TVM with attestation Evidence confirming its identity and the state of its firmware and configuration.

> Since Evidence claims can be appraised by 3rd party Verifiers external to the TVM, there is a need to standardize the representation of DA in Evidence to ensure interoperability.

Reviewer: ```suggestion
Since Evidence claims can be processed by 3rd party entities (e.g., Verifiers, Relying Parties) external to the TVM, there is a need to standardize the representation of DA-related information in Evidence to ensure interoperability.
```

This includes, for example, protection of device MMIO interfaces and device caches.
From a trust perspective, DA allows a device to be included in the TVM's Trusted Computing Base (TCB).
For the TVM to trust the device, the device must provide the TVM with attestation Evidence confirming its identity and the state of its firmware and configuration.

This document defines an attestation Evidence format for DA as an EAT {{-rats-eat}} profile.
The format is designed to be generic, extensible and architecture-agnostic.
Ongoing work on DA concentrates on PCIe devices that support the SPDM protocol {{-spdm}}, but other bus architectures and protocols are expected to be supported as the technology gains wider adoption.
As such, this document focuses on the formalization of an Evidence format for SPDM-compliant devices while leaving room for the definition of other Evidence formats such as Compute Express Link (CXL) and the Coherent Hub Interface (CHI).
This list is by no means exhaustive and is expected to expand.

# Conventions and Definitions

{::boilerplate bcp14-tagged}

# Device Assignment Token (DAT) Claims

> The Device Attestation Token (DAT) is the encompassing envelope for the individual device claims to be presented.

Reviewer: ```suggestion
The Device Assignment Token (DAT) is the encompassing envelope for the individual device claims to be presented.
```

> The Device Assignment Token (DAT) is an EAT that acts as the encompassing envelope for the individual device claims set to be presented.

Reviewer: ```suggestion
The Device Attestation Token (DAT) is the encompassing envelope for the individual device claims to be presented.
```

> A DAT can be used as a standalone entity but can also be embedded in a larger, platform-specific EAT using the `da-token` claim.

Reviewer: ```suggestion
A DAT can be used as a standalone entity but can also be embedded in a larger, platform-specific attestation token.
```
